reader/components/toolbar/Toolbar.jsx

```diff
@@ -154,19 +221,55 @@ const Toolbar = ({ viewer, eventBus }) => {
           type="button"
           className="btn"
           disabled={viewer.currentPageNumber <= 1}
-          onClick={() => --viewer.currentPageNumber}
+          onClick={() =>
+            dispatch({
+              type: 'change_input_number',
+              payload: {
+                inputNumber: --viewer.currentPageNumber,
+              },
+            })
+          }
         >
           <Icon iconType="left-arrow" />
         </button>
-        <div>
-          {viewer.currentPageNumber} / {viewer.pagesCount}
+        <div className="page-number-navigation">
+          <label className="m-0" htmlFor="page-number">
+            <span className="sr-only">Page number</span>
+            <input
+              type="text"
+              className="form-control input-change-page-number"
+              name="page-number"
+              value={state.inputNumber}
+              onChange={e => {
+                const pageNumber = Number(e.target.value)
+                if (Number.isNaN(pageNumber) || pageNumber > viewer.pagesCount)
+                  return
+
+                dispatch({
+                  type: 'change_input_number',
+                  payload: {
+                    inputNumber: e.target.value,
+                  },
+                })
+              }}
+              onBlur={handleBlurInput}
+            />{' '}
```

Comment:
I am wondering you are using controlled input. Let me play with it.

It's better to use browser API. You cannot avoid browser API anyway and you do not need to redo what browser can already do.

Reply:
It should work with `type=number` but I do not like this arrows at the right corner. Let's keep your solution.